agent/xds/endpoints.go

```diff
@@ -470,6 +482,21 @@ func makeLoadAssignment(clusterName string, endpointGroups []loadAssignmentEndpo
 	return cla
 }
 
+// This could be take Service instead of CheckServiceNode
+func makeAddressForService(node structs.CheckServiceNode, wan bool) *envoy_core_v3.Address {
+	var address *envoy_core_v3.Address
+	if node.Service.SocketPath != "" {
+		// TODO markan Decide what to do when both port and socket path are set; would like to warn but we don't have the logger available.
```

Comment:
You could promote this function and the relevant callers to have `func (s *Server)` receivers. I've done that before to get secret free loggers.